### DISCUSSION TITLE
Options for amsmath package

### DISCUSSION THREAD

**Discussion by matzebert** [jgm/pandoc#9606] [Q&A]
I would like to pass the option `intlimits` to the `amsmath` package when generating a PDF file via LuaLaTeX from a Markdown file. The command `\PassOptionsToPackage` in `header-includes` does not work since the `\usepackage{amsmath}` appears before the processing of the includes in the `default.latex` template. How can I pass the option without using a custom template?

**Comment by jgm** [comment]
You'd have to use a custom template for that, I think.
Or would it work to simply repeat the `\usepackage[your options]{amsmath}` in header-includes?  I don't know.

**Reply by matzebert** [reply]
This produces: `! LaTeX Error: Option clash for package amsmath.` Adding `\PassOptionsToPackage{intlimits}{amsmath}`, however, does not have any effect, because the `\usepackage{amsmath}` appears first in the LaTeX template.

Would it be an idea to have an "early hook" comparable to `header-includes` directly after the `\documentclass` for such cases?

**Reply by jgm** [reply]
We could, in principle, but I'd like to avoid the complexity and it's already easy to use a custom template.
You could even generate your custom template dynamically from the default one, if you're worried about maintaining it going forward.

Another option would be to add an `amsmath-options` variable.  I haven't looked at the options that are possible with amsmath, nor has anyone else requested this.

**Reply by matzebert** [reply]
After some research I have found out that `\PassOptionsToPackage{intlimits}{amsmath}` in a custom template does not work at all because `unicode-math` is loaded. With `\removenolimits{\int}` in `header-includes` I can achieve the desired result.

I think, however, that a _generic_ way to pass options to packages loaded in the default template might be a valuable feature for a future pandoc release. Introducing a separate options variable for each package might blow the template too much.
